src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2133,6 +2133,19 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 		(groupedByDisjointPartitionColumn || pushDownWindowFunction))
 	{
 		workerExtendedOpNode->havingQual = originalOpNode->havingQual;
+
+		/*
+		 * Postgres standard planner converts having qual node to a list of and
```

Comment:
We are doing this so that the worker query could be deparsed. This information (although correct) is irrelevant to why we are doing this.

Reply:
done!